### PR TITLE
Fix for issue #12897 (Changing to vi mode via config magic)

### PR DESCRIPTION
**Pull request by madbird1304**
Fix for #12897 

https://user-images.githubusercontent.com/18216480/131258901-c716cebe-9acd-4667-9d41-c8f209a50770.mp4

### CODE REVIEW

File: IPython/terminal/interactiveshell.py [hunk 1]
```diff
@@ -201,12 +201,10 @@ def _validate_editing_mode(self, proposal):
...
 
-
-    @observe('editing_mode')
-    def _editing_mode(self, change):
-        u_mode = change.new.upper()
+    @observe("editing_mode")
+    def _editing_mode_changed(self, change):
         if self.pt_app:
-            self.pt_app.editing_mode = u_mode
+            self.init_prompt_toolkit_cli()
```
[the remark this answers is not in the document]
Comment (madbird1304) [reply IPython/terminal/interactiveshell.py]: @Carreau , I found the specific cause of a problem. May I ask you to take a look? Thanks in advance!

**Comment by madbird1304** [comment]
@Carreau , or maybe @MrMino , may I ask you to take a look, please? Thanks!

**Comment by MrMino** [comment]
Will do as soon as I'll find the time to get back to OSS.

**Comment by Carreau** [comment]
Thanks for your patience, we are all relatively busy. 
The change looks straightforward to me, and if it fixes tings for you, let's try it.

**Comment by MrMino** [comment]
Hat tip for taking care of it, @Carreau.

Thanks for the contribution @madbird1304!

**Pull request merged**